docs/plugin_developers_guide/09_packaging.rst

```diff
@@ -0,0 +1,256 @@
+
+1. VERSION
+
+.. code-block:: console
+
+    $ cat VERSION
+    1.9.2
+
+2. fledge.version
+
+.. code-block:: console
+
+    $ cat fledge.version
+    fledge_version>=1.9
+
+3. requirements.sh
+
+.. code-block:: console
+
+    $ cat requirements.sh
+    #!/usr/bin/env bash
+    which apt >/dev/null 2>&1
+    if [ $? -eq 0 ]; then
+        sudo apt install -y libmodbus-dev
+    else
+        which yum >/dev/null 2>&1
+        if [ $? -eq 0 ]; then
+            sudo yum -y install epel-release libmodbus libmodbus-devel
+        fi
+    fi
+
+4. Description
+
+.. code-block:: console
+
+    $ cat Description
+    Fledge modbus plugin. Supports modbus RTU and modbus TCP.
+
+5. Package
+
+.. code-block:: console
+
+    $ cat Package
+    # A set of variables that define how we package this repository
+    #
+    plugin_name=modbus
+    plugin_type=south
+    plugin_install_dirname=ModbusC
+    plugin_package_name=fledge-south-modbus
+    additional_libs="usr/local/lib:/usr/local/lib/libsmod.so*"
+
+    # Now build up the runtime requirements list. This has 3 components
+    #   1. Generic packages we depend on in all architectures and package managers
+    #   2. Architecture specific packages we depend on
+    #   3. Package manager specific packages we depend on
+    requirements="fledge"
+
+    case "$arch" in
+        x84_64)
+            ;;
+        armv7l)
+            ;;
+        aarch64)
+            ;;
+    esac
+    case "$package_manager" in
+        deb)
+            requirements="${requirements},libmodbus-dev"
+            ;;
+        rpm)
+            requirements="${requirements},epel-release,libmodbus,libmodbus-devel"
+            ;;
+    esac
+
+.. note::
+    If your package is not supported for a specific platform then you must exit with exitcode 1
+
+6. service_notification.version
+
+.. code-block:: console
+
+    $ cat service_notification.version
+    service_notification_version>=1.9.2
+
+Common Additional Libraries Package
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+At the moment we have only two package exists.
+
+- **fledge-mqtt** which is packed with libpaho-mqtt library.
+- **fledge-gcp** which is packed with libjwt and libjansson libraries.
+
+If your plugin depends upon then update **requirements** variable in Package file instead of using *additional_libs* variable.
+
+Python based Plugins
+--------------------
+
+- VERSION.{PLUGIN_TYPE}.{PLUGIN_NAME}
+   Fledge and Plugin version number.
+- install_notes.txt (OPTIONAL)
+   If plugin requires any manual intervention like reboot or any setting which is required on the device directly. Therefore we need some notes to let user know what else is required to complete the installation of a plugin. And notes will be appeared at the end of plugin installation.
+- extras_install.sh (OPTIONAL)
+   This is just another shell script file to avoid package lock errors during package installation. We can write those steps here. Also make sure file should be in executable format.
```

Comment:
The *extras_install.sh* file is a shell script that is added to the package to allow for extra commands to be executed as part of the package installation. Not all plugins will require this file to be present and it can be omitted if there are no extra steps required on the installation.